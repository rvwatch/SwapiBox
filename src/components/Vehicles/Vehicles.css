.vehicles-container{
    margin: 120px auto auto auto;
    text-align: center;
  }