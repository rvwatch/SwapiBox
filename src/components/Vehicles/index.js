import React, { Component } from 'react';
import { getData, apiRoot } from '../apiCalls';
import Card from '../Cards';

class Vehicles extends Component {
  constructor() {
    super();

    this.state = {
      vehicles: []
    };
  }

  async componentDidMount() {
    const data = await getData(apiRoot + 'vehicles');
    const vehicles = await this.formatData(data);

    this.setState({ vehicles });
  }

  async formatData({ results }) {
    const unresolvedPromises = results.map(async vehicle => {
      //let homeworld = await getData(vehicle.homeworld);
      //let species = await getData(vehicle.species);

      return {
        title: vehicle.name,
        data1: `Model: ${vehicle.model}`,
        data2: `Class: ${vehicle.vehicle_class}`,
        data3: `Passengers: ${vehicle.passengers}`
      };
    });
    return await Promise.all(unresolvedPromises);
  }

  buildCards = data => {
    if (this.state.vehicles.length) {
      return data.map((vehicle, index) => (
        <Card
          type="vehicle"
          setFavorites={this.props.setFavorites}
          selected={vehicle}
          {...vehicle}
          key={index}
        />
      ));
    }
  };

  render() {
    const vehicle = this.buildCards(this.state.vehicles);
    return <section className="vehicles-container">{vehicle}</section>;
  }
}

export default Vehicles;
