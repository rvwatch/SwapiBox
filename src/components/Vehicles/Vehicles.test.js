import React from 'react';
import { shallow } from 'enzyme';
import Vehicles from './index';

describe('Vehicles ', () => {
  let wrapper;
  let mockVehicles;
  let mockSetFavs;
  beforeEach(() => {
    mockVehicles = [
      {
        data1: 'Homeworld: Tatooine',
        data2: 'Species: Human',
        data3: 'Population: 200000',
        favorite: true,
        title: 'Luke Skywalker'
      }
    ];
    window.fetch = jest.fn();
    mockSetFavs = jest.fn();
    wrapper = shallow(
      <Vehicles favorites={mockVehicles} setFavorites={mockSetFavs} />
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when given two cards', () => {
    mockVehicles = [
      {
        data1: 'Homeworld: Tatooine',
        data2: 'Species: Human',
        data3: 'Population: 200000',
        favorite: true,
        title: 'Luke Skywalker'
      },
      {
        data1: 'Homeworld: Naboo',
        data2: 'Species: Droid',
        data3: 'Population: 4500000000',

        favorite: true,
        title: 'R2-D2'
      }
    ];
    expect(wrapper).toMatchSnapshot();
  });


});
