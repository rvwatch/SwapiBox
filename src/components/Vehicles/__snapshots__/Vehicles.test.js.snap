// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    favorites={
        Array [
            Object {
              "data1": "Homeworld: Tatooine",
              "data2": "Species: Human",
              "data3": "Population: 200000",
              "favorite": true,
              "title": "Luke Skywalker",
            },
          ]
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "vehicles-container",
    },
    "ref": null,
    "rendered": null,
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "vehicles-container",
      },
      "ref": null,
      "rendered": null,
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Vehicles  renders correctly when given two cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    favorites={
        Array [
            Object {
              "data1": "Homeworld: Tatooine",
              "data2": "Species: Human",
              "data3": "Population: 200000",
              "favorite": true,
              "title": "Luke Skywalker",
            },
          ]
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "vehicles-container",
    },
    "ref": null,
    "rendered": null,
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "vehicles-container",
      },
      "ref": null,
      "rendered": null,
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
